FreeRTOS/Test/Target/tests/smp/schedule_highest_priority/schedule_highest_priority.c

```diff
@@ -0,0 +1,206 @@
+ */
+static void vPrvEverRunningTask( void * pvParameters );
+
+/**
+ * @brief Function that returns which index does the xCurrentTaskHandle match.
+ *        0 for T0, 1 for T1, -1 for not match.
+ */
+static int lFindTaskIdx( TaskHandle_t xCurrentTaskHandle );
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Handles of the tasks created in this test.
+ */
+static TaskHandle_t xTaskHanldes[ configNUMBER_OF_CORES ];
+
+/**
+ * @brief A flag to indicate if test case is finished.
+ */
+static BaseType_t xIsTestFinished = pdFALSE;
+/*-----------------------------------------------------------*/
+
+static int lFindTaskIdx( TaskHandle_t xCurrentTaskHandle )
+{
+    int i = 0;
+    int lMatchIdx = -1;
+
+    for( i = 0; i < configNUMBER_OF_CORES; i++ )
+    {
+        if( xCurrentTaskHandle == xTaskHanldes[ i ] )
+        {
+            lMatchIdx = i;
+            break;
+        }
+    }
+
+    return lMatchIdx;
+}
+/*-----------------------------------------------------------*/
+
+static void vPrvEverRunningTask( void * pvParameters )
+{
+    int i = 0;
+    int lCurrentTaskIdx = lFindTaskIdx( xTaskGetCurrentTaskHandle() );
+    eTaskState taskState;
+
+    /* Silence warnings about unused parameters. */
+    ( void ) pvParameters;
+
+    for( i = 0; i < configNUMBER_OF_CORES; i++ )
+    {
+        if( !xTaskHanldes[ i ] )
+        {
+            break;
+        }
+
+        taskState = eTaskGetState( xTaskHanldes[ i ] );
+
+        if( i <= lCurrentTaskIdx )
+        {
+            TEST_ASSERT_EQUAL_INT( eRunning, taskState );
+        }
+    }
+
+    if( lCurrentTaskIdx == configNUMBER_OF_CORES - 1 )
+    {
+        xIsTestFinished = pdTRUE;
+    }
+
+    for( ; ; )
+    {
+        /* Always running, put asm here to avoid optimization by compiler. */
+        __asm volatile ( "nop" );
+    }
+}
+/*-----------------------------------------------------------*/
+
+void Test_ScheduleHighestPirority( void )
+{
+    TickType_t xStartTick = xTaskGetTickCount();
+
+    /* Wait other tasks. */
+    while( xIsTestFinished == pdFALSE )
+    {
+        vTaskDelay( pdMS_TO_TICKS( 10 ) );
+
+        if( ( xTaskGetTickCount() - xStartTick ) / portTICK_PERIOD_MS >= TEST_TIMEOUT_MS )
+        {
+            break;
+        }
+    }
+
+    TEST_ASSERT_TRUE( xIsTestFinished == pdTRUE );
+}
+/*-----------------------------------------------------------*/
+
+/* Runs before every test, put init calls here. */
+void setUp( void )
+{
+    int i;
+    BaseType_t xTaskCreationResult;
+
+    /* Create configNUMBER_OF_CORES - 1 low priority tasks. */
+    for( i = 0; i < configNUMBER_OF_CORES; i++ )
+    {
+        xTaskCreationResult = xTaskCreate( vPrvEverRunningTask,
+                                           "EverRun",
+                                           configMINIMAL_STACK_SIZE * 2,
+                                           NULL,
+                                           configMAX_PRIORITIES - 1 - i,
```

Comment:
configMAX_PRIORITIES and configNUMBER_OF_CORES limitataion

Reply:
Added.
```
#if ( configMAX_PRIORITIES <= configNUMBER_OF_CORES )
    #error This test creates tasks with different priority, requires configMAX_PRIORITIES to be larger than configNUMBER_OF_CORES.
#endif /* if configNUMBER_OF_CORES != 2 */
```